tests/ui/linkage-attr/linkage-attr-mutable-static.rs

```diff
@@ -20,5 +20,6 @@ fn main() {
 
     unsafe {
         assert_eq!(WEAK_DEF, 0);
+        //~^ WARN creating a shared reference to mutable static is discouraged [static_mut_refs]
```

Comment:
Since this test is unrelated, let's fix (or `allow(..)` and mark `FIXME`) this and other similar uses instead.